connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -5883,4 +5883,65 @@ object functions {
   }
   // scalastyle:off line.size.limit
 
+  //////////////////////////////////////////////////////////////////////////////////////////////
+  // Conditional Functions
+  //////////////////////////////////////////////////////////////////////////////////////////////
+
+  /**
+   * If `predicate` evaluates to true, then returns `trueValue`; otherwise returns `falseValue`.
+   *
+   * @group conditional_funcs
+   * @since 3.5.0
+   */
+  def `if`(predicate: Column, trueValue: Column, falseValue: Column): Column =
+    Column.fn("if", predicate, trueValue, falseValue)
+
+  /**
+   * Returns `col2` if `col1` is null, or `col1` otherwise.
+   *
+   * @group conditional_funcs
+   * @since 3.5.0
+   */
+  def ifnull(col1: Column, col2: Column): Column = Column.fn("ifnull", col1, col2)
+
+  /**
+   * Returns true if `col` is not null, or false otherwise.
+   *
+   * @group conditional_funcs
+   * @since 3.5.0
+   */
+  def isnotnull(col: Column): Column = Column.fn("isnotnull", col)
+
+  /**
+   * Returns same result as the EQUAL(=) operator for non-null operands, but returns true if both
+   * are null, false if one of the them is null.
+   *
+   * @group conditional_funcs
+   * @since 3.5.0
+   */
+  def equal_null(col1: Column, col2: Column): Column = Column.fn("equal_null", col1, col2)
```

Comment:
I don't know the reason put the `EqualNull` into the group `misc_funcs`. It seems EqualNull should putted into group `predicate_funcs`. cc @cloud-fan @zhengruifeng

Reply:
I see a few functions in `FunctionRegistry` (and other places) were placed in wrong group. e.g. `Abs` should be in `math` instead of `misc non-aggregate functions`

Reply:
I think we should fix it together.

Reply:
So, let's put these functions in 'predicate_funcs' first? and later I will propose a new PR to fix grouping and other trivial issues. @zhengruifeng @beliefer

Reply:
I think it is fine, let's fix the grouping in separate PR